library/std/src/sys/pal/sgx/mod.rs

```diff
@@ -115,11 +115,14 @@ pub fn abort_internal() -> ! {
     abi::usercalls::exit(true)
 }
 
-// This function is needed by the panic runtime. The symbol is named in
+// This function is needed by libunwind. The symbol is named in
 // pre-link args for the target specification, so keep that in sync.
+// Note: contrary to the `__rust_abort` in `crate::rt`, this uses `no_mangle`
+//       because it is actually used from C code. Because symbols annotated with
+//       #[rustc_std_internal_symbol] get mangled, this will not lead to linker
+//       conflicts.
```

Comment:
Could they get different symbol names? Having the same extern signature reused but distinguished by attributes is a bit tricky.

Reply:
`__rust_abort` is the obvious name for the symbol shared between `panic_abort` and `std`, so I'm a bit hesitant about changing that. Changing the SGX symbol seems better, but it's a lot of effort as it requires changing libunwind as well; I don't really think it's worth it.

Reply:
Symbols marked as `#[rustc_std_internal_symbol]` don't really need the `__rust_` prefix anymore now that we do symbol mangling for them too.